### PR TITLE
fix: #MZI-131 fix returned mail with 0 recipient + fix move to trash

### PR DESCRIPTION
**Pull request by Matiyeu**
## Describe your changes

Fix status when a return mail request has only external senders. The status remained blocked in WAITING and the request did not complete.
The status now comes in REMOVED.

## Checklist tests

1. Send mail to external mail(s) ONLY
2. Call the mail back
3. Validate the mail request on the admin
4. The status will be in REMOVED instead of PROGRESS and request complete

## Issue ticket number and link

[#MZI-131](https://jira.support-ent.fr/browse/MZI-131)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by sjollois — APPROVED** [review]

File: zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java [hunk 1]
```diff
@@ -327,11 +330,12 @@ private void deleteMailFromZimbra(List<String> ids, UserInfos user, Handler<Eith
                             if (event.isRight()) {
                                 handler.handle(new Either.Right<>(event.right().getValue()));
                             } else {
-                                handler.handle(new Either.Left<>("[Zimbra] getReturnedMailsInfos : Error while deleting mails : " + event.left().getValue()));
+                                handler.handle(new Either.Left<>("[Zimbra] deleteMailFromZimbra : Error when deleting mails : " + event.left().getValue()));
                             }
                         });
                     } else {
                         log.error("Erreur lors du déplacement vers la corbeille du mail " + ids.get(0));
+                        handler.handle(new Either.Left<>("[Zimbra] deleteMailFromZimbra : Error when moving to trash : " + ids.get(0)));
```
Comment (JinDouang) [inline zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java]: A mettre dans log.error (base toi sur ton log error que tu avais fait la dernière fois avec gar-connector)

utilise le Either.Left avec `moveToTrash.left.getValue`

File: zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java [hunk 2]
```diff
@@ -327,11 +330,12 @@ private void deleteMailFromZimbra(List<String> ids, UserInfos user, Handler<Eith
                             if (event.isRight()) {
                                 handler.handle(new Either.Right<>(event.right().getValue()));
                             } else {
-                                handler.handle(new Either.Left<>("[Zimbra] getReturnedMailsInfos : Error while deleting mails : " + event.left().getValue()));
+                                handler.handle(new Either.Left<>("[Zimbra] deleteMailFromZimbra : Error when deleting mails : " + event.left().getValue()));
```
Comment (JinDouang) [inline zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java]: A mettre dans log.error (base toi sur ton log error que tu avais fait la dernière fois avec gar-connector)

utilise le Either.Left avec event.left.getValue

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java [hunk 3]
```diff
@@ -246,13 +246,16 @@ private void recursiveDeleteMailLoop(JsonArray returnedMails, JsonArray returned
...
                 this.recursiveDeleteMailLoop(returnedMails, returnedMailsStatut, index + 1, result);
             }
-            ;
         });
     }
 
     private void addDeleteFutures(JsonObject returnedMail, Handler<Either<String, JsonObject>> result) {
         JsonArray userIds = new JsonArray(returnedMail.getString(NOTIF_RECIPIENT));
         int j = 0;
+        if (userIds.size() == 0) {
```
Comment (JinDouang) [inline zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java]: Vérifie si c'est null et puis tu as la méthode `isEmpty`

```suggestion
        if (userIds == null || userIds.isEmpty()) {
```

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java [hunk 4]
```diff
@@ -327,11 +330,13 @@ private void deleteMailFromZimbra(List<String> ids, UserInfos user, Handler<Eith
                             if (event.isRight()) {
                                 handler.handle(new Either.Right<>(event.right().getValue()));
                             } else {
-                                handler.handle(new Either.Left<>("[Zimbra] getReturnedMailsInfos : Error while deleting mails : " + event.left().getValue()));
+                                log.error(String.format("[Zimbra@%s::deleteMailFromZimbra] Error when deleting mails: %s", this.getClass().getSimpleName(), event.left().getValue()));
+                                handler.handle(new Either.Left<>(String.format("[Zimbra@%s::deleteMailFromZimbra] Error when deleting mails: %s", this.getClass().getSimpleName(), event.left().getValue())));
```
Comment (JinDouang) [inline zimbra/src/main/java/fr/openent/zimbra/service/impl/ReturnedMailService.java]: Pas besoin d'envoyer le même log dans le handler, mets seulement le getValue
Les logs avec le format etc est plus pour les logs de vertx

```suggestion
                                                               handler.handle(event.left().getValue());

```

**Review by JinDouang — APPROVED** [review]

**Review by aymericdevriesere — APPROVED** [review]

**Pull request merged**